### PR TITLE
Better forum editor

### PR DESCRIPTION
**Pull request by easrng**
**Changes**

Adds an addon

**Reason for changes**

It makes the forum editor better

**Tests**

I tested it and it works. Fixed bug from last time.

### CODE REVIEW

File: addons/better-forum-editor/userscript.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    del(text) {
+      return `[s]${text}[/s]`;
+    },
+    link(href, title, text) {
+      return `[url=${href}]${text}[/url]`;
+    },
+    image(link) {
+      if (!isWhiteListed(link)) {
+        let l = new URL(link);
+        link = `https://secure.wikimedia.org/wikipedia/${l.hostname}%5C/..%5C${l.pathname.match(/^\/+(.+)$/)[1] || ""}`;
```
Comment (WorldLanguages) [inline addons/better-forum-editor/userscript.js:109]: This loophole has been fixed?

Comment (GrahamSH-LLK) [reply addons/better-forum-editor/userscript.js:109]: Yeah.

**Comment by WorldLanguages** [comment]
- Clicking the load image icon with no selected text throws an error in an alert box - I doubt that's intentional?
- In full screen mode (as well as split view mode), you likely want to hide the Scratch navbar so that it's not above the options bar of the markdown editor?

File: addons/better-forum-editor/userscript.js (hunk 1 above)
Comment (easrng) [reply addons/better-forum-editor/userscript.js:109]: yep

**Comment by JeffaloBob** [comment]
how does it currently work if another addon that edits the forum (namely the image uploader) is also enabled?

**Comment by WorldLanguages** [comment]
Oh, didn't notice that. Good luck making both compatible :c

**Comment by JeffaloBob** [comment]
> Oh, didn't notice that. Good luck making both compatible :c

make it so addon manifest can specify addons that aren't compatibile

**Comment by WorldLanguages** [comment]
@JeffaloBob No, they should just be compatible.

**Comment by YOYITsM3M8** [comment]
Does it include markup

**Comment by WorldLanguages** [comment]
Closing for now

**Pull request closed without merging**